File: app/src/main/java/com/jinhanexample/designPattern/adapter/java/operation/AddOperationAdapter.java
package com.jinhanexample.designPattern.adapter.java.operation;

public class AddOperationAdapter extends AbstractOperationTarget {
    protected OperationAdaptee operationAdaptee;

    public AddOperationAdapter(OperationAdaptee operationAdaptee) {
        this.operationAdaptee = operationAdaptee;
    }

    @Override
    public int operate(int firstNumber, int secondNumber) {
        return operationAdaptee.calculate(OperationAdaptee.ADD_OPERATION, firstNumber, secondNumber);
    }
}

File: app/src/main/java/com/jinhanexample/designPattern/adapter/kotlin/operation/AddOperationAdapter.kt
package com.jinhanexample.designPattern.adapter.kotlin.operation

class AddOperationAdapter(private var operationAdaptee: OperationAdaptee) :
    AbstractOperationTarget() {

    override fun operate(firstNumber: Int, secondNumber: Int): Int {
        return operationAdaptee.calculate(OperationAdaptee.Add_Operation, firstNumber, secondNumber)
    }

}